task_schedule3.pl

```diff
@@ -107,10 +111,14 @@
 	$_->{cmd} = $safe->reval(qq/"$_->{cmd}"/);
 
 	# If (after interpolation) the exec isn't an absolute path
-	# and there is a bin_dir defined, then prepend that to path
+	# and there is a bin_dir defined and there is an executable
+    # at bin_dir/cmd, then prepend that to path
 	if (not $_->{cmd} =~ m|\A \s* /|x and $opt{bin_dir}) {
-	    $_->{cmd} = "$opt{bin_dir}/$_->{cmd}";
-	}
+        my ($bin_dir_cmd, @cmd_args) = split(" ", "$opt{bin_dir}/$_->{cmd}");
```

Comment:
If there are @cmd_args, where do they go?  I don't see that handled.

Reply:
Yeah... cripes.  I just came to the same conclusion the hard way by testing and looking at logs in debug mode.

Reply:
Yeah, sorry I missed that in my review.  Though I note it seems to be working fine for arc which I thought was a stressing case (though maybe it doesn't use this bin_dir section).

Reply:
Clearly the test coverage is inadequate here to have caught this.  And of course in Python my editor would have complained that the new variable is never used.

Reply:
arc3 uses a better idiom, calling out the path instead of magically finding it in share.
```
      exec 10 : $ENV{SKA_SHARE}/arc3/get_iFOT_events.pl
      exec  5 : $ENV{SKA_SHARE}/arc3/get_web_content.pl
```
Zen of Python no. 2: Explicit is better than implicit!